src/main/java/org/opensearch/security/ssl/http/netty/Netty4HttpRequestHeaderVerifier.java

```diff
@@ -83,28 +75,16 @@ public void channelRead0(ChannelHandlerContext ctx, DefaultHttpRequest msg) thro
         ctx.channel().attr(IS_AUTHENTICATED).set(Boolean.FALSE);
 
         final Netty4HttpChannel httpChannel = ctx.channel().attr(Netty4HttpServerTransport.HTTP_CHANNEL_KEY).get();
-        String rawPath = SecurityRestUtils.path(msg.uri());
-        String path = RestUtils.decodeComponent(rawPath);
-        Matcher matcher = PATTERN_PATH_PREFIX.matcher(path);
-        final String suffix = matcher.matches() ? matcher.group(2) : null;
-        if (API_AUTHTOKEN_SUFFIX.equals(suffix)) {
```

Comment:
is this logic being moved to somewhere else?

Reply:
The way this is handled is that the SAML Authenticator throws a `SecurityRequestChannelUnsupported` on this endpoint and the header verifier knows how to handle this type of exception. @peternied introduced this error as a general way that other endpoints could follow a similar pattern if need be. The `/_plugins/_security/api/authtoken` endpoint consumes a SAML Response (XML) in the body of the request and uses that to authenticate. It is the only endpoint that requires body for authn, all other endpoints only need the headers of the request.